lib/remote-connector.js

```diff
@@ -91,14 +129,56 @@ function createProxyMethod(Model, remotes, remoteMethod) {
       callback = args.pop();
     }
 
+    // Take care not to overwrite method args
+    var numberOfDefaultArgs = remoteMethod.accepts.length - headerArgs.length;
+    headerArgs.forEach(function(headerArg, idx) {
+      args[idx + numberOfDefaultArgs] = headerArg;
+    });
+
+    if (auth && auth.accessTokenId) {
+      // Find correct index for auth header
+      remoteMethod.accepts.forEach(function(accept, idx) {
+        if (accept.name === 'authorization') {
+          args[idx] = auth.accessTokenId;
+        }
+      });
+    }
+
+    // Make sure gaps in array are numbered.
+    for (var i = 0, argsLength = args.length; i < argsLength; i++) {
+      args[i] = args[i] || void 0;
+    }
+
     if (remoteMethod.isStatic) {
       return remotes.invoke(remoteMethod.stringName, args, callback);
     }
 
     var ctorArgs = [this.id];
     return remotes.invoke(remoteMethod.stringName, ctorArgs, args, callback);
-  }
+  };
 }
 
-function noop() {
+function interceptUserRemotes(remotes, auth) {
+  remotes.after('User.logout', function(ctx, next) {
+    auth.clearUser();
+    auth.clearStorage();
+    next();
+  });
+
+  remotes.before('User.login', function(ctx, next) {
+    auth.rememberMe = ctx.req.body.rememberMe || false;
```

Comment:
This doesn't seem to be the right place for this. I think we should remove the (although helpful) `User.login` hooks entirely and leave this up to app developers, tooling, loopback, etc. The APIs are available to do this outside of this module so I think that is the first step in the right direction.

I'd like to land the header / auth support without adding opinionated implementations of the storing the auth headers. We can always do that later if we need to.

@BerkeleyTrue what do you think about opening a separate issue to discuss support for the opinionated `User.login / logout` hooks? Perhaps in the LoopBack repo.